tests/src/main.rs

```diff
@@ -60,11 +60,21 @@ fn main() {
     let codegen_backend_path = find_rustc_codegen_spirv();
     let libs = build_deps(&deps_target_dir, &codegen_backend_path);
 
+    run_mode(
+        "compile-fail",
```

Comment:
Aren't non-UI tests deprecated? If you want `//~ ERROR` annotations, you can still use them with UI, but IMO it's really valuable to keep the output around, to be able to catch it changing in unexpected ways.